solr/core/src/java/org/apache/solr/cli/SolrCLI.java

```diff
@@ -91,12 +91,45 @@ public class SolrCLI implements CLIO {
           .build();
   public static final Option OPTION_SOLRURL =
       Option.builder("solrUrl")
-          .argName("HOST")
+          .argName("SOLR_URL")
           .hasArg()
           .required(false)
           .desc(
               "Base Solr URL, which can be used to determine the zkHost if that's not known; defaults to: "
-                  + getDefaultSolrUrl()
+                  + getDefaultSolrUrl(null)
+                  + '.')
+          .build();
+  public static final Option OPTION_SOLRPORT =
+      Option.builder("p")
+          .longOpt("port")
+          .argName("PORT")
+          .hasArg()
+          .required(false)
+          .desc(
+              "Solr Port, which will be used, along with SOLR_TOOL_HOST and SOLR_URL_SCHEME, to determine the Solr URL to connect to if a solrURL is not provided; defaults to: "
+                  + getSolrPort(null)
+                  + '.')
+          .build();
+  public static final Option OPTION_SOLRHOST =
+      Option.builder("host")
+          .longOpt("host")
+          .argName("SOLR_HOST")
+          .hasArg()
+          .required(false)
+          .desc(
+              "Solr Host, which will be used, along with SOLR_PORT and SOLR_URL_SCHEME, to determine the Solr URL to connect to if a solrURL is not provided; defaults to: "
+                  + getSolrHost(null)
+                  + '.')
+          .build();
+  public static final Option OPTION_URLSCHEME =
+      Option.builder("urlScheme")
+          .longOpt("url-scheme")
+          .argName("urlScheme")
+          .hasArg()
+          .required(false)
+          .desc(
+              "Solr URL Scheme, which will be used, along with SOLR_TOOL_HOST and SOLR_PORT, to determine the Solr URL to connect to if a solrURL is not provided; defaults to: "
```

Comment:
SOLR_HOST instead of SOLR_TOOL_HOST?  again, I don't quite get the difference ;-).